datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -295,12 +306,17 @@ def event(self, event):
                         'Error decoding unicode field `{}` to utf-8 encoded string, cannot submit event'.format(key)
                     )
                     return
+
         if event.get('tags'):
             event['tags'] = self._normalize_tags_type(event['tags'])
         if event.get('timestamp'):
             event['timestamp'] = int(event['timestamp'])
         if event.get('aggregation_key'):
             event['aggregation_key'] = ensure_unicode(event['aggregation_key'])
+
+        if self.__NAMESPACE__:
+            event.setdefault('source_type_name', self.__NAMESPACE__)
```

Comment:
This is the only bit that would introduce a change in behavior. I don't really object to it, it's just worth noting.